src/main/webapp/resources/js/pages/projects/metadata/MetadataTemplatesList.jsx

```diff
@@ -25,16 +43,62 @@ import { unwrapResult } from "@reduxjs/toolkit";
  * @constructor
  */
 export function MetadataTemplatesList({ projectId }) {
-  const { canManage } = useSelector((state) => state.project);
   const { templates, loading } = useSelector((state) => state.templates);
+  const { canManage } = useSelector((state) => state.project);
+
+  const { fields } = useSelector((state) => state.fields);
+
   const dispatch = useDispatch();
   const [BASE_URL] = React.useState(() =>
     setBaseUrl(`/projects/${projectId}/metadata-templates`)
   );
 
+  const [templatesModified, setTemplatesModified] = React.useState([{}]);
+
   /**
-   * This crates the "actions" that appear at the right of every row in
-   * the table: field count, download template, and remove template (if applicable).
+   * Set default metadata template for project.
+   *
+   * @param {number} templateId - identifier for the metadata template to set as default
+   */
+  const setDefaultTemplate = async (templateId) => {
+    await dispatch(setDefaultTemplateForProject({ projectId, templateId }))
+      .then(unwrapResult)
+      .then(({ message }) => {
+        notification.success({ message });
+      })
+      .catch((message) => notification.error({ message }));
+  };
+
+  React.useEffect(() => {
+    if (templates != null && fields != null) {
```

Comment:
I feel this should be handled in the templateSlice, what do you think?

Reply:
Unfortunately as per our discussion we can't do that currently due to a race condition. We can revisit this later

Reply:
So I thought about this last night as it was frustrating me.  I still feel it should be created inside the `templatesSlice`.  Here is how I think you should handle it:

1.  In `templateSlice`, set your `initialState.template` to be: `[{/* your allFieldsTemplate object */}]}` without the fields attribute, or the default attribute (see other comment).

You have a couple options on how to set the fields:

1.  In `MetadataTemplateList`  you actually don't need the useEffect.  The only thing you need that for is for the setting the number of fields on the template.  You can check to see if it is the default template being rendered and just get the fields count right there from the state -> `fields` array;
2.  Or, which might be better, you can get the `dispatch` function within an `createAsyncThunk` (also found in the `payloadCreator` second object.  When you have recieved all the fields you could dispatch a call to template and give them the fields for the all template (since it will be there already in the initial state).

Reply:
Great solution. Thanks! Updated in a90712821956a7f57805affd2f5dcd54932b0984. Went with Option 2 for setting the fields with dispatch.